File: LeetCode-September-Challenge/ConcatenationofConsecutiveBinaryNumbers.java
class Solution {
    public int concatenatedBinary(int n) {
        long val = 0;
        int i = 1;
        while(i <= n){
            val = ((val << (1 + (int)(Math.log(i)/Math.log(2)))) % 1000000007 + i) % 1000000007;
            ++i;
        }
        return (int)val;
    }
}
public class ConcatenationofConsecutiveBinaryNumbers {
    public static void main(String[] args) {
        Solution s = new Solution();
        System.out.println(s.concatenatedBinary(3));
    }
}

File: Year-2022/LeetCode-September-Challenge/ConcatenationofConsecutiveBinaryNumbers.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
private:
    const int MOD = 1e9 + 7;
public:
    int concatenatedBinary(int n) {
        long long int val = 0;
        int i = 1;
        while(i <= n){
            val = ((val << (1 + int(log2(i)))) % MOD + i) % MOD;
            ++i;
        }
        return val;
    }
};
int main(){
    Solution s;
    cout<<s.concatenatedBinary(3)<<endl;
    return 0;
}